src/R/Editor/Impl/Documentation/RoxygenBlock.cs

```diff
@@ -0,0 +1,106 @@
+
+            Variable v;
+            IFunctionDefinition fd = FunctionDefinitionExtensions.FindFunctionDefinition(textBuffer, ast, position, out v);
+            if (fd != null && v != null && !string.IsNullOrEmpty(v.Name)) {
+
+                int definitionStart = Math.Min(v.Start, fd.Start);
+                int blockPosition = GetRoxygenBlockPosition(snapshot, currentLine.LineNumber, definitionStart);
+                if (blockPosition < 0) {
+                    string lineBreak = currentLine.GetLineBreakText();
+                    if (string.IsNullOrEmpty(lineBreak)) {
+                        lineBreak = "\n";
+                    }
+                    string block = GenerateRoxygenBlock(v.Name, fd, lineBreak);
+                    if (block.Length > 0) {
+                        textBuffer.Replace(new Span(currentLine.Start, currentLine.Length), block);
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        private static int GetRoxygenBlockPosition(ITextSnapshot snapshot, int currentlineNumber, int definitionStart) {
+            var line = snapshot.GetLineFromPosition(definitionStart);
+            for (int i = line.LineNumber - 1; i >= 0; i--) {
+                if (currentlineNumber >= 0 && i == currentlineNumber) {
+                    // Skip line where user just typed ###
+                    continue;
+                }
+                line = snapshot.GetLineFromLineNumber(i);
+                if (line.Length > 0) {
+                    var lineText = line.GetText().TrimStart();
+                    if (lineText.Length > 0) {
+                        if (lineText.StartsWith("#'", StringComparison.Ordinal)) {
+                            return line.Start + 1;
+                        } else {
+                            break;
+                        }
+                    }
+                }
+            }
+            return -1;
+        }
+
+        private static string GenerateRoxygenBlock(string functionName, IFunctionDefinition fd, string lineBreak) {
+            var sb = new StringBuilder();
+
+            IFunctionInfo fi = fd.MakeFunctionInfo(functionName);
+            if (fi != null && fi.Signatures.Count > 0) {
+
+                sb.Append(Invariant($"#' Title{lineBreak}"));
+                sb.Append(Invariant($"#'{lineBreak}"));
+
+                int length = sb.Length;
+                foreach (var p in fi.Signatures[0].Arguments) {
+                    if (!string.IsNullOrEmpty(p.Name)) {
+                        sb.Append(Invariant($"#' @param {p.Name}{lineBreak}"));
+                    }
+                }
+
+                if (sb.Length > length) {
+                    sb.Append(Invariant($"#'{lineBreak}"));
+                }
+
+                sb.Append(Invariant($"#' @return{lineBreak}"));
+                sb.Append(Invariant($"#' @export{lineBreak}"));
```

Comment:
Should we always be inserting `@export`? My understanding is that it will actually translate to the function being exported from the package (i.e. it's more than documentation); but are all functions with Roxygen comments normally exported?